src/Razor/src/Microsoft.VisualStudio.RazorExtension/SyntaxVisualizer/SyntaxVisualizerControl.xaml.cs

```diff
@@ -48,6 +49,15 @@ public SyntaxVisualizerControl()
             InitializeComponent();
 
             InitializeRunningDocumentTable();
+        }
+
+        [MemberNotNull(nameof(_codeDocumentProvider), nameof(_textDocumentFactoryService), nameof(_joinableTaskFactory))]
```

Comment:
Ah wasn't aware of `MemberNotNull`. Coolbeans, I assume it means if you call the method then the specified members can be considered as not null?

Reply:
Yep, exactly that. It's the first time I've been able to use it, because normally I want it on `Initialize` methods that are called automatically, but the compiler doesn't know that.